hooks/cmd/configure/configure.go

```diff
@@ -0,0 +1,159 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2017-2021 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+        "log/syslog"
+	"os"
+	"strconv"
+
+	"github.com/canonical/app-service-configurable/hooks"
+)
+
+var log syslog.Writer
+
+func handleVal(p string, k string, v interface{}, flatConf map[string]interface{}) {
+	var mk string
+
+	// top level keys don't include "env", so no separator needed
+	if p == "" {
+		mk = k
+	} else {
+		mk = fmt.Sprintf("%s.%s", p, k)
+	}
+
+	log.Info(fmt.Sprintf("handleVal: mk: %s", mk))
+
+	switch t := v.(type) {
+	case string:
+		log.Info(fmt.Sprintf("ADDING %s=%s to flatConf", k, t))
+		flatConf[mk] = t
+	case bool:
+		log.Info(fmt.Sprintf("ADDING %s=%v to flatConf", k, t))
+		flatConf[mk] = strconv.FormatBool(t)
+	case float64:
+		log.Info(fmt.Sprintf("ADDING %s=%v to flatConf", k, t))
+		flatConf[mk] = strconv.FormatFloat(t, 'f', -1, 64)
+	case map[string]interface{}:
+		log.Info(fmt.Sprintf("FOUND AN OBJECT"))
+
+		for k, v := range t {
+			handleVal(mk, k, v, flatConf)
+		}
+	default:
+		log.Err("I DON'T KNOW!!!!")
+	}
+}
+
+func handleSvcConf(env string) {
+	log.Info(fmt.Sprintf("edgex-asc:configure:handleSvcConf config is %s", env))
+
+	if env == "" {
+		return
+	}
+
+	var m map[string]interface{}
+	var flatConf = make(map[string]interface{})
+	//flatM = make(map[string]interface{})
+
+	err := json.Unmarshal([]byte(env), &m)
+	if err != nil {
+		log.Err(fmt.Sprintf("edgex-asc:configure:handleSvcConf: failed to unmarshall env; %v", err))
+		return
+	}
+
+	for k, v := range m {
+		handleVal("", k, v, flatConf)
+	}
+
+	path := fmt.Sprintf("%s/config/res/service.env", hooks.SnapDataE)
+
+	f, err := os.OpenFile(path, os.O_WRONLY|os.O_CREATE|os.O_TRUNC, 0644)
+	if err != nil {
+		log.Err(fmt.Sprintf("edgex-asc:configure:handleSvcConf: can't open %s - %v", path, err))
+		os.Exit(1)
+	}
+
+	defer f.Close()
+
+	log.Info(fmt.Sprintf("edgex-asc:configure:handleSvcConf about write %s", path))
+	for k, v := range flatConf {
+		log.Info(fmt.Sprintf("%s=%v", k, v))
+		_, err := f.WriteString(fmt.Sprintf("export %s=%s\n", hooks.ConfToEnv[k], v))
```

Comment:
this could just be

```suggestion
		_, err := fmt.Fprintf(f, "export %s=%s\n", hooks.ConfToEnv[k], v)
```

and if you use a bytes.Buffer{} you can just use `b` as the io.Writer instead of `f`. and then at the end of this you would do:

```go
ioutil.WriteFile(path, b.Bytes(), 0644)
```
or whatever permissions are appropriate